src/post/post.repository.knex.ts

```diff
@@ -1,38 +1,155 @@
-                title: post.title,
-                excerpt: post.excerpt,
-                content: post.content,
-                url: post.url?.href,
-                image_url: post.imageUrl?.href,
-                published_at: post.publishedAt,
-                in_reply_to: post.inReplyTo?.id,
-                thread_root: post.threadRoot?.id,
-                like_count: 0,
-                repost_count: 0,
-                reply_count: 0,
-                reading_time_minutes: post.readingTime,
-                ap_id: post.apId.href,
-            });
-            // Hacks? Mutate the Post so `isNew` returns false.
-            (post as any).id = id;
-        } else {
-            // Handle updates
+        const transaction = await this.db.transaction();
+
+        try {
+            const potentiallyNewLikes = post.getPotentiallyNewLikes();
+
+            if (post.isNew) {
+                const [id] = await transaction('posts').insert({
+                    uuid: post.uuid,
+                    type: post.type,
+                    audience: post.audience,
+                    author_id: post.author.id,
+                    title: post.title,
+                    excerpt: post.excerpt,
+                    content: post.content,
+                    url: post.url?.href,
+                    image_url: post.imageUrl?.href,
+                    published_at: post.publishedAt,
+                    in_reply_to: post.inReplyTo?.id,
+                    thread_root: post.threadRoot?.id,
+                    like_count: potentiallyNewLikes.length,
+                    repost_count: 0,
+                    reply_count: 0,
+                    reading_time_minutes: post.readingTime,
+                    ap_id: post.apId.href,
+                });
+
+                if (potentiallyNewLikes.length > 0) {
+                    const likesToInsert = potentiallyNewLikes.map(
+                        (accountId) => ({
+                            account_id: accountId,
+                            post_id: id,
+                        }),
+                    );
+
+                    await transaction('likes').insert(likesToInsert);
+                }
+                // Hacks? Mutate the Post so `isNew` returns false.
+                (post as any).id = id;
+            } else {
+                if (potentiallyNewLikes.length > 0) {
+                    const likesToInsert = potentiallyNewLikes.map(
+                        (accountId) => ({
+                            account_id: accountId,
+                            post_id: post.id,
+                        }),
+                    );
+
+                    await transaction('likes')
+                        .insert(likesToInsert)
+                        .onConflict()
+                        .ignore();
+
+                    const [[{ insertedRows }]] = await transaction.raw(
+                        'SELECT ROW_COUNT() as insertedRows',
+                    );
```

Comment:
🤔 Is this because you don't know if the like was inserted or not from the above query?

Reply:
Yeah exactly - we use `ON CONFLICT IGNORE` so we don't know how many rows are inserted